### ISSUE TITLE
rt-event-panel table layout

### ISSUE THREAD

**Issue by mattttt** [raintank/grafana#309]
Core issue:

Link to panel:  https://app.raintank.io/dashboard/file/rt-events.json
Dev path to panel: public/plugins/raintank/panels/raintankEventsPanel/module.html

The Litmus Events panel works quite well at narrow widths (using flexbox), which is extremely valuable in certain dashboards:

![screenshot 2015-07-07 16 16 07](https://cloud.githubusercontent.com/assets/2886187/8556456/1ae182b0-24c5-11e5-9a38-6f4791ae17b2.png)

but it's not as nice when it's a full width table view:

![screenshot 2015-07-07 16 16 50](https://cloud.githubusercontent.com/assets/2886187/8556667/8267aa26-24c6-11e5-9d87-8381b1fc1b7c.png)

with that, I propose we use mediaqueries to address the width of the panel (if possible), and use a traditional table view when feasible that looks more like this:

![table-view-mockup](https://cloud.githubusercontent.com/assets/2886187/8556709/c8377d42-24c6-11e5-8506-03d74a15506e.png)

**bonus points**

Figure out what to do at these widths, which are a bit awkward. The endpoint and collector names can be any length, which means that hard breakpoints arent always going to be the right time to switch:

![screenshot 2015-07-07 16 16 36](https://cloud.githubusercontent.com/assets/2886187/8556742/03d66188-24c7-11e5-9b4d-7664b490f4c9.png)
![screenshot 2015-07-07 16 16 26](https://cloud.githubusercontent.com/assets/2886187/8556743/03d962f2-24c7-11e5-8dc7-183dbf61eac3.png)

**Comment by mattttt** [comment]
This is really looking good - I think we can adjust the column widths to give more space to Check Error Description to avoid breakpoints like these:

![screenshot 2015-07-09 09 09 49](https://cloud.githubusercontent.com/assets/2886187/8595978/964c2b84-261a-11e5-85c7-2d1e6266f1b5.png)
![screenshot 2015-07-09 09 09 32](https://cloud.githubusercontent.com/assets/2886187/8595979/964c2c60-261a-11e5-897a-ee48b2b2ece6.png)
![screenshot 2015-07-09 09 09 12](https://cloud.githubusercontent.com/assets/2886187/8595980/964ce862-261a-11e5-8f83-44878274b167.png)

**Comment by mattttt** [comment]
@nchristus I know you're on vacation, so no rush on response. Noticing that it no longer is responsive for the smaller widths. How can we make it the best of both of these worlds?

**Comment by mattttt** [comment]
Also, this is what it's looking like at wide layouts, super nice. 

![screenshot 2015-07-21 17 28 58](https://cloud.githubusercontent.com/assets/2886187/8813030/0e1935c8-2fce-11e5-8a11-f7d59eb5484b.png)

**Comment by mattttt** [comment]
Keeping branch: nchristus-rt-event-panel-table-layout alive for the moment for future changes, but as it stands currently, has been moved into master.

**Comment by mattttt** [comment]
Rolled back because it was breaking like whoah:

![screen_shot_2015-07-24_at_11 50 36_am_1024](https://cloud.githubusercontent.com/assets/2886187/8879696/dfa954ea-31e7-11e5-890a-e5f28ee3ef28.png)

**Comment by nchristus** [comment]
Updated the layout to accommodate the additional column.

These data table layouts are tricky, especially when responsive.

It's looking much better now :)

**Comment by mattttt** [comment]
Have to test this. Will come back tomorrow/Friday.

**Comment by nopzor1200** [comment]
Cool here's to hoping it looks better - @woodsaj I'd still like to resolve #44 so that we can use this panel in more places (and allow users to use it wherever they want)

**Comment by mattttt** [comment]
After chatting and exchanging screenshots, we realized that the issue seems to be related to the menu being open. When closed, it works great and steps down responsively. @nchristus working on a fix.

**Comment by mattttt** [comment]
@nchristus This is sooo close. It's much nicer than what's their name, so I'm going to merge this branch. 

That said, the last thing that I think needs to happen to to consolidate the endpoint and collector items into a single line at the lowest breakpoints. Since these are handled semantically as <li> items, I dont know how to make both appear on the same line:

##### What it is:
<img width="573" alt="screenshot 2015-08-05 23 10 29" src="https://cloud.githubusercontent.com/assets/2886187/9103057/ae14b19c-3bc7-11e5-94ae-847e9a2fbc29.png">

##### What I'd like it to be:
<img width="597" alt="screenshot 2015-08-05 23 09 36" src="https://cloud.githubusercontent.com/assets/2886187/9103064/b1be50b4-3bc7-11e5-8621-69035e733fa5.png">

**Comment by mattttt** [comment]
Here's a quick and dirty animated gif showing the different breakpoints for the branch Im going to merge:

![event-panel](https://cloud.githubusercontent.com/assets/2886187/9103106/26510f34-3bc8-11e5-8189-ac55ae5186b6.gif)

**Comment by mattttt** [comment]
Actually, Im trying to create a new pull request, and it is saying that it's up to date with master. @woodsaj Was this already merged in?

Also, on production there is some formatting weirdness that Im unable to replicate on this branch:
<img width="1292" alt="screenshot 2015-08-05 23 35 18" src="https://cloud.githubusercontent.com/assets/2886187/9103310/b18dd526-3bca-11e5-98c1-e8b7878b9af5.png">

**Comment by woodsaj** [comment]
@mattttt  what branch?  the nchristus-rt-event-panel-table-layout branch mentioned earlier in this issue hasn't had a commit for 15days

**Comment by mattttt** [comment]
@woodsaj sorry, neglected to mention it. The actual branch is: `nchristus-collector-dropdown`

The other one can be deleted, it's stale.

**Comment by mattttt** [comment]
@woodsaj: @nchristus just brought to my attention that I inadvertently merged this during my changes on Monday in PR https://github.com/raintank/grafana/pull/389/commits

You can cease to research, totally my fault

**Comment by mattttt** [comment]
Waiting on @ctdk to updated dev-portal and see if this issue still exists.

**Comment by woodsaj** [comment]
this should be in dev portal now, and probably also prod.

**Comment by mattttt** [comment]
Looks solid. Closing.
